app/src/main/java/org/openobservatory/ooniprobe/activity/runtests/RunTestsActivity.kt

```diff
@@ -0,0 +1,189 @@
+				putSerializable(TESTS, testSuites as Serializable)
+			})
+		}
+	}
+
+	override fun onCreate(savedInstanceState: Bundle?) {
+		super.onCreate(savedInstanceState)
+		binding = ActivityRunTestsBinding.inflate(layoutInflater)
+		setContentView(binding.getRoot())
+
+		setSupportActionBar(binding.toolbar)
+		supportActionBar?.setDisplayHomeAsUpEnabled(true)
+
+		activityComponent?.inject(this)
+
+		val testSuites: List<AbstractSuite>? = intent.extras?.getParcelableArrayList(TESTS)
+		testSuites?.let { ts ->
+			val tsGroups: List<GroupItem> = ts.map { testSuite ->
+				return@map testSuite.runTestsGroupItem(preferenceManager)
+			}
+
+			mAdapter = RunTestsExpandableListViewAdapter(this, tsGroups, viewModel)
+
+			binding.expandableListView.setAdapter(mAdapter)
+			for (i in 0 until mAdapter.groupCount) {
+				binding.expandableListView.expandGroup(i)
+			}
+			binding.selectAll.setOnClickListener { onSelectAllClickListener() }
+
+			binding.selectNone.setOnClickListener { onSelectNoneClickListener() }
+
+			viewModel.selectedAllBtnStatus.observe(this, this::selectAllBtnStatusObserver)
+
+			binding.bottomBar.setOnMenuItemClickListener { menuItem ->
+				onMenuItemClickListener(menuItem)
+			}
+		} ?: run {
+			finish()
+		}
+
+	}
+
+	override fun onCreateOptionsMenu(menu: Menu): Boolean {
+		val inflater: MenuInflater = menuInflater
+		inflater.inflate(R.menu.close, menu)
+		return true
+	}
+
+	override fun onOptionsItemSelected(item: MenuItem): Boolean {
+		return when (item.itemId) {
+			R.id.close_button -> {
+				finish()
+				true
+			}
+
+			else -> super.onOptionsItemSelected(item)
+		}
+	}
+
+	private fun onMenuItemClickListener(menuItem: MenuItem): Boolean {
+		return when (menuItem.itemId) {
+			R.id.runButton -> {
+				val selectedChildItems: List<String> = getChildItemsSelectedIdList()
+				if (selectedChildItems.isNotEmpty()) {
+					val testSuitesToRun = getGroupItemsAtLeastOneChildEnabled().map { groupItem ->
+						return@map AbstractSuite.getTestSuiteByName(groupItem.name).dynamicTestSuite(
+							nettests = groupItem.nettests.filter { nattest -> nattest.selected }
+						)
+					}
+					RunningActivity.runAsForegroundService(
+						this@RunTestsActivity,
+						java.util.ArrayList(testSuitesToRun),
+						{ finish() },
+						preferenceManager
+					)
+
+				}
+				true
+			}
+
+			else -> false
+		}
+	}
+
+	// TODO(aanorbel) Update button color from theme
+	private fun selectAllBtnStatusObserver(selectAllBtnStatus: String?) {
+		if (!TextUtils.isEmpty(selectAllBtnStatus)) {
+			when (selectAllBtnStatus) {
+				SELECT_ALL -> {
+					binding.selectNone.isActivated = true
+					binding.selectAll.isActivated = false
+				}
+
+				SELECT_NONE -> {
+					binding.selectNone.isActivated = true
+					binding.selectAll.isActivated = false
+				}
+
+				SELECT_SOME -> {
+					binding.selectNone.isActivated = true
+					binding.selectAll.isActivated = true
+				}
+			}
+			mAdapter.notifyDataSetChanged()
```

Comment:
We're not generally using the `mFoo` convention for internals. Are we planning on using it more frequently? If not, maybe I'd just avoid the `mFoo` convention. (Unless there's a reason for using it that I don't see.)